app/src/examples/ShareablesExample.tsx

```diff
@@ -159,10 +193,118 @@ function DataViewDemo() {
   return <Button title="DataView" onPress={handlePress} />;
 }
 
+function FreezingShareables() {
+  const navigation =
+    useNavigation<NativeStackNavigationProp<StackParams, 'Freezing'>>();
+
+  return (
+    <View style={styles.container}>
+      <Button
+        title="Try modify converted array"
+        onPress={tryModifyConvertedArray}
+      />
+      <Button
+        title="Try modify converted remote function"
+        onPress={tryModifyConvertedRemoteFunction}
+      />
+      <Button
+        title="Try modify converted host object"
+        onPress={tryModifyConvertedHostObject}
+      />
+      <Button
+        title="Try modify converted plain object"
+        onPress={tryModifyConvertedPlainObject}
+      />
+      <Button
+        title="Try modify converted regex literal"
+        onPress={tryModifyConvertedRegexLiteral}
+      />
+      <Button
+        title="Try modify converted RegExp instance"
+        onPress={tryModifyConvertedRegexInstance}
+      />
+      <Button
+        title="Try modify converted ArrayBuffer"
+        onPress={tryModifyConvertedArrayBuffer}
+      />
+      <Button
+        title="Try modify converted Int32Array"
+        onPress={tryModifyConvertedInt32Array}
+      />
+      <View style={styles.bar} />
+      <Button title="Go to creating" onPress={() => navigation.goBack()} />
+    </View>
+  );
+}
+
+function tryModifyConvertedArray() {
+  const arr = [1, 2, 3];
+  makeShareableCloneRecursive(arr);
+  arr[0] = 2; // should warn beacuse it's frozen
+}
+
+function tryModifyConvertedRemoteFunction() {
+  const foo = () => {};
+  foo.bar = 1;
+  makeShareableCloneRecursive(foo);
+  foo.bar = 2; // should warn because it's frozen
+}
+
+function tryModifyConvertedHostObject() {
+  const hostObject = TurboModuleRegistry.get('Clipboard');
+  if (!hostObject) {
+    console.warn('No host object found.');
+    return;
+  }
+  makeShareableCloneRecursive(hostObject);
+  // @ts-expect-error
+  hostObject.prop = 2; // shouldn't warn because it's not frozen
+}
+
+function tryModifyConvertedPlainObject() {
+  const obj = {
+    a: 1,
+  };
+  makeShareableCloneRecursive(obj);
+  obj.a = 2; // should warn because it's frozen
+}
+
+function tryModifyConvertedRegexLiteral() {
+  const regexLiteral = /a/;
+  makeShareableCloneRecursive(regexLiteral);
+  // @ts-expect-error
+  regexLiteral.regexProp = 2; // shouldn't warn because it's not frozen
```

Comment:
Let's just use `obj` and `prop` to make the code more repeatable and easier to spot differences.
```suggestion
  const obj = /a/;
  makeShareableCloneRecursive(obj);
  // @ts-expect-error
  obj.prop = 2; // shouldn't warn because it's not frozen
```